src/region/RegionAnnotations.tsx

```diff
@@ -60,10 +60,11 @@ export default class RegionAnnotations extends React.PureComponent<Props, State>
         setStatus(CreatorStatus.started);
     };
 
-    handleStop = (shape: Rect): void => {
+    handleStop = (shape: Rect | null): void => {
```

Comment:
Would adding an `onAbort` prop and `handleAbort` handler simplify things here?

Reply:
Could you describe what you had in mind for that?

Reply:
Would probably make the intention more clear. In the `RegionCreator`, perhaps in `stopDraw`, if there is no shape, we could call `onAbort` instead and the `handleAbort` is very clearly just resetting `setStaged` and `setStatus` --probably could even call `resetCreator` which would already emits the `cancel` event

Reply:
Cool thanks for the suggestion, will address.